dango/types/src/oracle/price.rs

```diff
@@ -26,9 +32,20 @@ impl PrecisionlessPrice {
 }
 
 impl PrecisionedPrice {
+    /// Returns the number of decimal places of the token that is used to convert
+    /// the price from its smallest unit to a humanized form. E.g. 1 ATOM
+    /// is 10^6 uatom, so the precision is 6.
     pub fn precision(&self) -> u8 {
         self.precision.into_inner()
     }
+
+    /// Returns the value of a given unit amount. E.g. if this Price represents
+    /// the price in USD of one ATOM, then this function will return the value in USD of
+    /// the given number of uatom.
+    pub fn value_of_unit_amount(&self, unit_amount: Uint128) -> Udec128 {
+        self.humanized_price * Udec128::new(*unit_amount.inner())
```

Comment:
Instead of `Udec128::new(*unit_amount.inner())` use `Udec128::raw(unit_amount)`?

Reply:
But that's not the same? `Udec128::new(42)` gives `"42.0"` whereas `Udec128::raw(10)` gives `0.00000000000000042`?